webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala

```diff
@@ -363,60 +365,75 @@ class BuildBundleSequence(
     val name = Promiser(policy.ruleName + "/" + policy.directiveName)
 
     // and for now, all bundle get the same reportKey
```

Comment:
All the following code is just a rewrite of the same algo in a failing for loop returning `PureResult` since at that point we should not modify anymore any variable value. If we don't have what we expect, it's a problem that should be exposed.